File: scala_api/src/main/scala/application/models/NumberColumnsSummary.scala
package application.models

import spray.json.{DefaultJsonProtocol, RootJsonFormat}

case class NumberColumnsSummary(
  name: String,
  count: Long,
  mean: Double,
  stdDev: Double,
  min: Double,
  percentage25: Double,
  percentage50: Double,
  percentage75: Double,
  max: Double
)

trait NumberColumnsSummaryJsonProtocol extends DefaultJsonProtocol {
  implicit val numberColumnsSummaryFormat: RootJsonFormat[NumberColumnsSummary] = jsonFormat9(NumberColumnsSummary)
}

File: python_api/Models/NumberColumnsSummary.py
class NumberColumnsSummary:
    def __init__(self, count, name, mean, stdDev, min,
                 percentage25, percentage50, percentage75, max):
        self.count = count
        self.name = name
        self.mean = mean
        self.stdDev = stdDev
        self.min = min
        self.percentage25 = percentage25
        self.percentage50 = percentage50
        self.percentage75 = percentage75
        self.max = max
